modin/pandas/dataframe.py

```diff
@@ -2541,6 +2541,10 @@ def __setitem__(self, key, value):
             self.insert(loc=len(self.columns), column=key, value=value)
             return
 
+        if isinstance(key, list) and is_list_like(value):
+            self.loc[slice(None), key] = value
```

Comment:
@dchigarev have you thought about using `loc/iloc` functions in `__setitem__` function, does it make sense for me to continue working in this direction?

Reply:
I think using `loc/iloc` may be a good approach if it supports forwarding `slice(None)` or `range(0, len(df))` row indexer right to the `qc.write_items`. If it converts the indexer to a list of positions (`[0, 1, 2, ..., len(df)]` along the way to `qc.write_items` it would be much less performance beneficial, as the performance-wise complex [`modin_frame._get_dict_of_block_index`](https://github.com/modin-project/modin/blob/97769988a6f19e4b76f34238c97bf159ee7626a5/modin/core/dataframe/pandas/dataframe/dataframe.py#L1103) function would need to sequentially process a `len(df)` indexer in case of a list-like one (for slice-like indexer it has [a fast-path](https://github.com/modin-project/modin/blob/97769988a6f19e4b76f34238c97bf159ee7626a5/modin/core/dataframe/pandas/dataframe/dataframe.py#L1132))